### ISSUE TITLE
Backup consistently failing because page title is too long

### ISSUE THREAD

**Issue by RoamHenry** [MatthieuBizien/roam-to-git#79]
**Describe the bug**
My backups now fail every single time.

**To Reproduce**

2020-12-14 00:56:32.821 | DEBUG    | roam_to_git.scrapping:download_rr_archive:76 - Closing browser markdown
2020-12-14 00:56:32.868 | DEBUG    | roam_to_git.scrapping:download_rr_archive:78 - Closed browser markdown
2020-12-14 00:56:32.869 | DEBUG    | roam_to_git.scrapping:scrap:255 - Scrapping finished
2020-12-14 00:56:32.916 | DEBUG    | roam_to_git.fs:save_markdowns:50 - Saving markdown to /home/runner/work/PersonalGraph/PersonalGraph/markdown
2020-12-14 00:56:32.998 | ERROR    | __main__:<module>:33 - An error has been caught in function '<module>', process 'MainProcess' (2787), thread 'MainThread' (140253104490304):
Traceback (most recent call last):
> File "/opt/hostedtoolcache/Python/3.8.6/x64/bin/roam-to-git", line 33, in <module>
    sys.exit(load_entry_point('roam-to-git==0.1', 'console_scripts', 'roam-to-git')())
    │   │    └ <function importlib_load_entry_point at 0x7f8f3708c4c0>
    │   └ <built-in function exit>
    └ <module 'sys' (built-in)>
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/roam_to_git/__main__.py", line 82, in main
    save_markdowns(git_path / "markdown", raws)
    │              │                      └ {'October 4th, 2020.md': '- [[Curiosity]] [[Science]] of [[Social Connection]]\n    - [[Stanford Medicine]] [[Center for Comp...
    │              └ PosixPath('/home/runner/work/PersonalGraph/PersonalGraph')
    └ <function save_markdowns at 0x7f8f328a3670>
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/roam_to_git/fs.py", line 57, in save_markdowns
    with dest.open("w", encoding="utf-8") as f:
         │    │                              └ <_io.TextIOWrapper name='/home/runner/work/PersonalGraph/PersonalGraph/markdown/shame.md' mode='w' encoding='utf-8'>
         │    └ <function Path.open at 0x7f8f36d72a60>
         └ PosixPath("/home/runner/work/PersonalGraph/PersonalGraph/markdown/Like optical illusions, intellectual illusions can trick us...
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/pathlib.py", line 1221, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
           │  │    │     │     │          │         │       └ None
           │  │    │     │     │          │         └ None
           │  │    │     │     │          └ 'utf-8'
           │  │    │     │     └ -1
           │  │    │     └ 'w'
           │  │    └ PosixPath("/home/runner/work/PersonalGraph/PersonalGraph/markdown/Like optical illusions, intellectual illusions can trick us...
           │  └ <built-in function open>
           └ <module 'io' from '/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/io.py'>
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/pathlib.py", line 1077, in _opener
    return self._accessor.open(self, flags, mode)
           │    │              │     │      └ 438
           │    │              │     └ 524865
           │    │              └ PosixPath("/home/runner/work/PersonalGraph/PersonalGraph/markdown/Like optical illusions, intellectual illusions can trick us...
           │    └ <member '_accessor' of 'Path' objects>
           └ PosixPath("/home/runner/work/PersonalGraph/PersonalGraph/markdown/Like optical illusions, intellectual illusions can trick us...

OSError: [Errno 36] File name too long: "/home/runner/work/PersonalGraph/PersonalGraph/markdown/Like optical illusions, intellectual illusions can trick us into thinking something that isn't actually there or true. As with optical illusions, even when we know they are there we still have to actively override our default perception to get to the truth behind the illusion..md"
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.6/x64/bin/roam-to-git", line 33, in <module>
    sys.exit(load_entry_point('roam-to-git==0.1', 'console_scripts', 'roam-to-git')())
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/loguru/_logger.py", line 1149, in catch_wrapper
    return function(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/roam_to_git/__main__.py", line 82, in main
    save_markdowns(git_path / "markdown", raws)
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/site-packages/roam_to_git/fs.py", line 57, in save_markdowns
    with dest.open("w", encoding="utf-8") as f:
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/pathlib.py", line 1221, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
  File "/opt/hostedtoolcache/Python/3.8.6/x64/lib/python3.8/pathlib.py", line 1077, in _opener
    return self._accessor.open(self, flags, mode)
OSError: [Errno 36] File name too long: "/home/runner/work/PersonalGraph/PersonalGraph/markdown/Like optical illusions, intellectual illusions can trick us into thinking something that isn't actually there or true. As with optical illusions, even when we know they are there we still have to actively override our default perception to get to the truth behind the illusion..md"
Error: Process completed with exit code 1.

**Expected behavior**
Backup to run and complete as expected.

**Please complete the following information:**
- OS: MacOS
- Do you use Github Action? Yes
- Do you use multiple Roam databases? No
- Does roam-to-git use to work for you? When precisely did it stopped to work? Yes. It stopped working after I made a page (one of my zettels) with a long title.
- Does some backup runs are still working? No

**Additional context**
I am using the Zettelkasten method which results in pages with longer names. Curious how I can resolve this because just removing the page or changing the name is not an option (if I keep up with Zettelkasten in Roam)

**Comment by everruler12** [comment]
Shorten the page title for `Like optical illusions, intellectual illusions can trick us...`. You probably accidentally tagged an entire sentence or paragraph as a page.

**Comment by RoamHenry** [comment]
I did, in fact, intentionally tag two sentences. This is part of the Zettelkasten workflow and I would like to find a mechanism to do backups where my workflow and the needs for that workflow are upheld. I notated that in the **Additional Context** section at the bottom.

Is there a way to implement backups with longer page names?

**Comment by everruler12** [comment]
Oh, I see. The markdown file is named after the page title, so it can't have reserved characters (as per #54 ) or a length greater than 255 characters. You would have to somehow disable markdown backups. This isn't an issue with roam-to-git backup, but with Roam's markdown backup. If you manually export as markdown within Roam, you won't be able to extract it if the title is too long.

A couple other suggestions:

- Put the full sentences onto the page, and shorten the title with ellipses, or a short name for the zettel concept.
- Use block references (( )) for long Zettels.

**Comment by RoamHenry** [comment]
Thanks for the clarity. The first solution won't work in the desired workflow, but I'm curious to better understand your suggestion for the second solution.

For the multi-sentence zettel, I create the zettel with double brackets so that I can easily see all the mentions of that zettel on it's page. There is nothing on that page other than the linked (and unlinked) references, and it's useful for quickly browsing elsewhere.

If I have the two sentences as a standalone block, and then reference that block in other locations, then I have to remember where that original block came from to be able to find it and all associated references to that block, no?

In short, the desired workflow decouples the zettel from it's source page and allows it to live on it's own. Later, I can add further comments and thoughts on that zettel's page as I further flesh out the idea. I could be wrong, but I think using block references would break that by forcing me to go back to the source page. Am I missing something?

If you can show a gif of what you mean that may help me understand better.

**Comment by everruler12** [comment]
I was suggesting putting the full text of the Zettel as a block in its standalone page, then give the Zettel a short descriptive name for the title to reference. Or you can reference the block and see all of its references.
![2020 12 17 104058](https://user-images.githubusercontent.com/64292121/102523125-60c6bd80-4054-11eb-8b5c-1cbc579ee271.gif)

**Comment by RoamHenry** [comment]
Thanks for the gif and I understand what you mean now.

I'll play around with this and see if it'll work for my purposes. My
intuition is that it'll add 1-3 clicks for every discovery step on every
zettel, thereby breaking the flow state of simply creating them in line as
standalone pages, but I'll try it to see how it works.

Lame that this is even an issue. I'll also reach out to the Roam team and
see if the limit on 255 characters for markdown export is changeable. If
that limitation goes away, then the git backup will work natively again
with no additional changes, right?

**Comment by everruler12** [comment]
The 255 character limit is imposed by Ubuntu, the Linux OS that this backup uses. Windows OS has a similar limit, while MacOS is like 1024. The Roam team could truncate page titles of markdown exports. However, the downsides of this are the possibility of filename collisions (though unlikely at that length), and losing data (the full page title, which would then need changed in all the other pages that reference it.) The best solution, I think, is for @MatthieuBizien to add an option that makes markdown exports optional, keeping JSON and EDN backups.

**Comment by MatthieuBizien** [comment]
This should be fixed by #82.  Feel free to re-open the issue if that's not the case.

**Comment by Stvad** [comment]
My backups still fail when I have pages with names that are too long 🙁
